src/game.py

```diff
@@ -0,0 +1,284 @@
+# Modified by Daniel Jauregui for study purposes
+
+import random
+
+from utils.limits import Limits
+
+
+class Game(Limits):
+
+    def __init__(self, min_threshold=20, max_threshold=30):
+        self.max_threshold = max_threshold
+        self.min_threshold = min_threshold
+        self.hints = []
+        self.board = []
+
+    def generate_game(self):
+        """Call to make_puzzle with a empty board to generate game
+        with level defined and it will return a array e.g.:
+        Array: [4, 9, 3, None, 5, ...., 8, 8, None, None, 1]
+        Where the position are from 0 to 80
+
+        return -- Array e.g.: [4, 9, 3, None, 5, ...., 8, 8, None, None, 1]
+        """
+        self.board = self.make_puzzle(self.get_solution([None] * 81))
+        return self.board
+
+    def make_puzzle(self, board):
+        """Get the solved board and remove the range of values
+         defined for level, the random position values will be
+         replaced for None.
+
+        Keyword arguments:
+        board -- Array with all values solved for sudoku.
+        return -- Array e.g.: [4, 9, 3, None, 5, ...., 8, 8, None, None, 1]
+        """
+        level = random.randint(self.min_threshold, self.max_threshold)
+        init_level = 0
+        while init_level < level:
+            position = random.randint(0, 80)
+            if board[position] is not None:
+                self.hints.append((position, board[position]+1))
+                board[position] = None
+                init_level += 1
+
+        for position in xrange(81):
+            if board[position] is None:
+                board[position] = 0
+            else:
+                board[position] += 1
+        return board
+
+    def get_solution(self, board):
+        """Get an empty array with 81 positions and call
+        solve_board method to get a solved sudoku.
+
+        Keyword arguments:
+        board -- Array e.g.: [None, None,..., None, None]
+        return  -- Array e.g.: [4, 9, 3, 4, 5, ...., 8, 7, 4, 8, 1]
+        """
+        return self.solve_board(board)[1]
+
+    def solve_board(self, original):
+        """Get an empty array (it can be resolve the puzzle)
+        and create an arrays of board in order to get an unique solution
+
+        Keyword arguments:
+        original -- Array of Arrays e.g.: [[None, None,..., None, None]]
+        return  -- Array e.g.: [4, 9, 3, 4, 5, ...., 8, 7, 4, 8, 1]
+        """
+        board = list(original)
+        guesses = self.deduce(board)
+        if guesses is None:
+            return [], board
+        track = [(guesses, 0, board)]
+        return self.solve_next(track)
+
+    def solve_next(self, remembered):
+        """Get a board and guesses required if game is empty,
+        it will create all game until a board does not contains None values
+
+        Keyword arguments:
+        remembered -- Contain all boards guesses, it is an Array e.g.: [[None, None,..., None, None]]
```

Comment:
your description mention that is an array but the example is an array of arrays is this correct?